### PR TITLE
`azurerm_api_management` - fix `subnet_id` case sensitivity

### PR DESCRIPTION
**Pull request by teowa**
Resolves https://github.com/hashicorp/terraform-provider-azurerm/issues/18399

read the `subnet_id` , and format it  before write to the state.

### CODE REVIEW

File: internal/services/apimanagement/api_management_resource.go [hunk 1]
```diff
@@ -1732,7 +1733,8 @@ func flattenApiManagementVirtualNetworkConfiguration(input *apimanagement.Virtua
 	virtualNetworkConfiguration := make(map[string]interface{})
 
 	if input.SubnetResourceID != nil {
-		virtualNetworkConfiguration["subnet_id"] = *input.SubnetResourceID
+		subnetId, _ := networkParse.SubnetID(*input.SubnetResourceID)
```
Comment (katbyte) [inline internal/services/apimanagement/api_management_resource.go:1736]: can we nil check this deference?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/apimanagement/api_management_resource.go [hunk 2]
```diff
@@ -1724,18 +1735,22 @@ func flattenApiManagementProtocolsCustomProperties(input map[string]*string) []i
...
 	if input == nil {
-		return []interface{}{}
+		return []interface{}{}, nil
 	}
 
 	virtualNetworkConfiguration := make(map[string]interface{})
 
 	if input.SubnetResourceID != nil {
-		virtualNetworkConfiguration["subnet_id"] = *input.SubnetResourceID
+		subnetId, err := networkParse.SubnetID(*input.SubnetResourceID)
```
Comment (katbyte) [inline internal/services/apimanagement/api_management_resource.go:1746]: should this not be `SubnetIDInsensitively` ?

**Review by katbyte — CHANGES_REQUESTED** [review]

**Review by tombuildsstuff — APPROVED** [review]
LGTM providing the tests pass 👍

**Comment by teowa** [comment]
image

The failed test also fail in main branch.

**Pull request merged**